src/App.tsx

```diff
@@ -1,61 +1,121 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/user';
+
+const findUserById = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) as User
+);
+
+const allTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId),
+}));
+
+export const App: React.FC<{}> = () => {
+  const [title, setTitle] = useState('');
+  const [currUser, selectUser] = useState('');
+  const [todos, setTodos] = useState(allTodos);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const getMaxId = () => {
```

Comment:
I advice you to use another method to find the max id, because sort is a mutating method, and it shouldn't be mutating the state like that. Check this out, should be cleaner:

```ts
return Math.max(...todos.map(todo => todo.id)) + 1;

Reply:
agree with @UmizDemud, that's critical one